### PR TITLE
Don't sync secrets before server is up.

### PR DESCRIPTION
**Pull request by rohinb2**

### CODE REVIEW

**Comment by rohinb2** [comment]
* **#1364** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1364?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1362** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1362?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1361** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1361?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1360** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1360?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1314** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1314?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1347** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1347?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1389** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1389?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rohinb2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: runhouse/resources/hardware/cluster.py [hunk 1]
```diff
@@ -1237,6 +1229,15 @@ def _start_or_restart_helper(
             self._start_ray_workers(DEFAULT_RAY_PORT, env=self.default_env)
 
         self.put_resource(self.default_env)
+
+        if self._default_env.secrets:
```
Comment (carolineechen) [inline runhouse/resources/hardware/cluster.py:1233]: probably use `if default_env and default_env.secrets` instead, `self._default_env` can be None

Comment (rohinb2) [reply runhouse/resources/hardware/cluster.py]: yep

**Review by carolineechen — APPROVED** [review]

**Comment by rohinb2** [comment]
### Merge activity

* **Nov 11, 1:38 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/run-house/runhouse/1389).

**Pull request merged**